File: Caronae/MenuViewController.m
#import "MenuViewController.h"
#import "ProfileViewController.h"
#import "UIImageView+crn_setImageWithURL.h"
#import "WebViewController.h"
#import "Caronae-Swift.h"

@interface MenuViewController ()
@property (nonatomic) NSString *photoURL;
@end

@implementation MenuViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.navigationItem.titleView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"NavigationBarLogo"]];
    
    [self updateProfileFields];
    
    [NSNotificationCenter.defaultCenter addObserver:self selector:@selector(updateProfileFields) name:CaronaeDidUpdateUserNotification object:nil];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    User *user = UserService.instance.user;
    NSString *userPhotoURL = user.profilePictureURL;
    if (userPhotoURL.length > 0 && ![userPhotoURL isEqualToString:self.photoURL]) {
        self.photoURL = userPhotoURL;
        [self.profileImageView crn_setImageWithURL:[NSURL URLWithString:userPhotoURL]];
    }
}

- (void)dealloc {
    [NSNotificationCenter.defaultCenter removeObserver:self];
}

- (void)updateProfileFields {
    User *user = UserService.instance.user;
    self.profileNameLabel.text = user.name;
    self.profileCourseLabel.text = user.course.length > 0 ? [NSString stringWithFormat:@"%@ | %@", user.profile, user.course] : user.profile;
}

#pragma mark - Navigation

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([segue.identifier isEqualToString:@"ViewProfile"]) {
        ProfileViewController *vc = segue.destinationViewController;
        vc.user = UserService.instance.user;
    }
    else if ([segue.identifier isEqualToString:@"About"]) {
        WebViewController *vc = segue.destinationViewController;
        vc.page = WebViewAboutPage;
    }
    else if ([segue.identifier isEqualToString:@"TermsOfUse"]) {
        WebViewController *vc = segue.destinationViewController;
        vc.page = WebViewTermsOfUsePage;
    }
}

- (void)openRidesHistory {
    [self performSegueWithIdentifier:@"RidesHistory" sender:nil];
}

@end

File: Caronae/MenuViewController.swift
import UIKit

class MenuViewController: UIViewController {
    
    @IBOutlet weak var profileImageView: UIImageView!
    @IBOutlet weak var profileNameLabel: UILabel!
    @IBOutlet weak var profileCourseLabel: UILabel!
    
    var photoURL: String!
    
    deinit {
        NotificationCenter.default.removeObserver(self)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        navigationController?.view.backgroundColor = .white
        navigationItem.titleView = UIImageView(image: UIImage(named: "NavigationBarLogo"))
        
        updateProfileFields()
        
        NotificationCenter.default.addObserver(self, selector:#selector(updateProfileFields), name: .CaronaeDidUpdateUser, object: nil)
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        if let user = UserService.instance.user,
            let userPhotoURL = user.profilePictureURL,
            !userPhotoURL.isEmpty,
            userPhotoURL != photoURL {
            
            photoURL = userPhotoURL
            profileImageView.crn_setImage(with: URL(string: userPhotoURL))
        }
    }
    
    @objc func updateProfileFields() {
        guard let user = UserService.instance.user else {
            return
        }
        
        profileNameLabel.text = user.name
        profileCourseLabel.text = user.course.isEmpty ? user.profile : String(format: "%@ | %@", user.profile, user.course)
    }
    
    
    // MARK: Navigation
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "ViewProfile" {
            let vc = segue.destination as! ProfileViewController
            vc.user = UserService.instance.user
        } else if segue.identifier == "About" {
            let vc = segue.destination as! WebViewController
            vc.page = .aboutPage
        } else if segue.identifier == "TermsOfUse" {
            let vc = segue.destination as! WebViewController
            vc.page = .termsOfUsePage
        } else if segue.identifier == "FAQ" {
            let vc = segue.destination as! WebViewController
            vc.page = .FAQPage
        }
    }
    
    func openRidesHistory() {
        performSegue(withIdentifier: "RidesHistory", sender: nil)
    }
}
